.travis.yml

```diff
@@ -129,6 +129,11 @@ matrix:
         - nvm install
         - ./bin/yarn install --pure-lockfile
 
+    - python: 3.7
+      name: 'Linter for Python 3'
+      install: pip install flake8
+      script: flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
```

Comment:
@cclauss Why are you explicitly selecting these four? We also don't really need `--count --show-source --statistics`, reason being it makes the output pretty verbose.

Reply:
F9 enables all F9xx tests.

https://flake8.pycqa.org/en/latest/user/error-codes.html

On the flake8 test selection, this PR does _not_ focus on "_style violations_" (the majority of flake8 error codes that [__psf/black__](https://github.com/psf/black) can autocorrect).  Instead these tests are focus on runtime safety and correctness:
* E9 tests are about Python syntax errors usually raised because flake8 can not build an Abstract Syntax Tree (AST).  Often these issues are a sign of unused code or code that has not been ported to Python 3.  These would be compile-time errors in a compiled language but in a dynamic language like Python they result in the script halting/crashing on the user.
* F63 tests are usually about the confusion between identity and equality in Python.  Use ==/!= to compare str, bytes, and int literals is the classic case.  These are areas where __a == b__ is True but __a is b__ is False (or vice versa).  Python >= 3.8 will raise SyntaxWarnings on these instances.
* F7 tests logic errors and syntax errors in type hints
* F82 tests are almost always _undefined names_ which are usually a sign of a typo, missing imports, or code that has not been ported to Python 3.  These also would be compile-time errors in a compiled language but in Python a __NameError__ is raised which will halt/crash the script on the user.

I would usually reject the PR if one of these errors was present so the size of helpful output is not a concern.  If these test pass then the output is one line which can be eliminated by removing --count.

Reply:
Thanks for elaborating! I guess what I'm really asking here is are `E9,F63,F7,F82` in the default ignore list (and that'd be why you're selecting them explicitly here)? I couldn't find the default ignore list after some casual googling.

Reply:
Thiese are in the select list (not ignore list) because they find issues of runtime safety.  You can find them in [Node.js](https://github.com/nodejs/node-gyp/blob/master/.travis.yml#L91), [GitHub Actions CI](https://github.com/actions/starter-workflows/blob/master/ci/python-package.yml#L28), [Internet Archive](https://github.com/internetarchive/openlibrary/blob/master/Makefile#L96) so this was not a project-specific choice.  I hope they help.

Reply:
Correct.  If you have a pre-commit hooks or editors that runs black/yapf/autopep8 then just run `flake8 .` but if your contributors don’t have some automation then they will quickly get tired of their PRs being flagged for whitespace issues.  The Bxxx errors must come from https://github.com/PyCQA/bandit

Reply:
B-something is from sentry-flake8.

Reply:
> Correct. If you have a pre-commit hooks or editors that runs black/yapf/autopep8 then just run `flake8 .` but if your contributors don’t have some automation then they will quickly get tired of their PRs being flagged for whitespace issues. The Bxxx errors must come from https://github.com/PyCQA/bandit

Very good point, although I think it would be better to emphasize in both dev docs and possibly in a pull request template to run the linter. We also do have pre-commit in place, which is installed via `make bootstrap`, but the linter hasn't been moved there yet for... reasons. I'll keep this in mind.